modules/025-static-routing-manager/images/agent/pkg/controller/routes_reconciler_agent.go

```diff
@@ -0,0 +1,572 @@
+				return reconcile.Result{}, nil
+			}
+
+			if nrt.Spec.NodeName != cfg.NodeName {
+				log.Debug(fmt.Sprintf("[NRTReconciler] This request is not intended(by spec.nodeName) for our node (%v)", cfg.NodeName))
+				return reconcile.Result{}, nil
+			}
+
+			if nrt.Generation == nrt.Status.ObservedGeneration && nrt.DeletionTimestamp != nil {
+				cond := FindStatusCondition(nrt.Status.Conditions, v1alpha1.ReconciliationSucceed)
+				if cond.Status == metav1.ConditionTrue {
+					log.Debug(fmt.Sprintf("[NRTReconciler] There's nothing to do"))
+					return reconcile.Result{}, nil
+				}
+			} else {
+				tmpNRT := new(v1alpha1.NodeRoutingTable)
+				*tmpNRT = *nrt
+				err := SetStatusConditionPending(ctx, cl, log, tmpNRT)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("[NRTReconciler] error"))
+				}
+			}
+
+			// ============================= main logic start =============================
+			// Declaring variables
+			globalDesiredRouteEntryMap := make(RouteEntryMap)
+			actualRouteEntryMap := make(RouteEntryMap)
+			deletedNRTRouteEntryMaps := make(map[string]RouteEntryMap)
+			erasedNRTRouteEntryMaps := make(map[string]RouteEntryMap)
+			nrtK8sResourcesMap := make(map[string]*v1alpha1.NodeRoutingTable)
+			nrtReconciliationStatusMap := make(map[string]NRTReconciliationStatus)
+			specNeedToUpdate := make(map[string]bool)
+			shouldRequeue := false
+
+			// Getting all the NodeRoutingTable associated with our node
+			nrtList := &v1alpha1.NodeRoutingTableList{}
+			err = cl.List(ctx, nrtList, client.MatchingLabels{nodeNameLabel: cfg.NodeName})
+			if err != nil && !errors2.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("[NRTReconciler] unable to list NodeRoutingTable for node %s", cfg.NodeName))
+				return reconcile.Result{RequeueAfter: cfg.RequeueInterval * time.Second}, err
+			}
+
+			// Getting all routes from our node
+			actualRouteEntryMap, err = getActualRouteEntryMapFromNode()
+			if err != nil {
+				log.Error(err, fmt.Sprintf("[NRTReconciler] unable to get Actual routes from node"))
+				return reconcile.Result{RequeueAfter: cfg.RequeueInterval * time.Second}, err
+			}
+
+			for _, nrt := range nrtList.Items {
+				// Gathering facts
+				log.Debug(fmt.Sprintf("[NRTReconciler] Starting gather facts about nrt %v", nrt.Name))
+				// Filling nrtK8sResourcesMap[nrt.Name] and nrtReconciliationStatusMap[nrt.Name]
+				tmpNrt := nrt
+				tmpNrt.Status.ObservedGeneration = nrt.Generation
+				nrtK8sResourcesMap[nrt.Name] = &tmpNrt
+				nrtReconciliationStatusMap[nrt.Name] = NRTReconciliationStatus{IsSuccess: true}
+				specNeedToUpdate[nrt.Name] = false
+
+				if nrt.DeletionTimestamp != nil {
+					log.Debug(fmt.Sprintf("[NRTReconciler] NRT %v is marked for deletion", nrt.Name))
+					var tmpREM RouteEntryMap
+					if len(deletedNRTRouteEntryMaps[nrt.Name]) == 0 {
+						tmpREM = make(RouteEntryMap)
+					} else {
+						tmpREM = deletedNRTRouteEntryMaps[nrt.Name]
+					}
+					for _, route := range nrt.Spec.Routes {
```

Comment:
The hash calculation move to AppendR() func.

Reply:
`AppendR` already includes the hash calculation. In this case, the hash is calculated to verify the presence in the `actualRouteEntryMap`. In any case, after moving the check for deleted routes on the node to the part where the deletion occurs, additional hash calculation is not necessary.